src/main/java/com/kizerov/caloriescalculator/service/CalculatorService.java

```diff
@@ -1,6 +1,60 @@
 package com.kizerov.caloriescalculator.service;
 
+import com.kizerov.caloriescalculator.model.CaloriesSummaryDto;
+import com.kizerov.caloriescalculator.model.Food;
+import com.kizerov.caloriescalculator.model.Meal;
+import com.kizerov.caloriescalculator.model.User;
+import com.kizerov.caloriescalculator.repository.MealRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 
+import java.time.LocalDate;
+import java.util.Collections;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+@Transactional
 public class CalculatorService implements ICalcService {
 
+    private final MealRepository mealRepository;
+    private final UserService userService;
+    private final FoodsService foodsService;
+
+    public CaloriesSummaryDto showStatistics(String email, LocalDate startDate, LocalDate endDate) {
+
+        List<Meal> result = mealRepository.findAllByUserEmailAndDate(email, startDate, endDate)
+                .orElse(Collections.emptyList());
+
+        return CaloriesSummaryDto.builder()
+                .startDate(startDate)
+                .endDate(endDate)
+                .totalCalories(result.stream()
+                        .mapToInt(eatenCalories ->
+                                eatenCalories.getWeight() * eatenCalories.getFood().getCaloriesPer100Gram() / 100)
+                        .sum())
+                .meals(result)
+                .build();
+    }
+
+    public void addTodayMeal(String userEmail, String productName, int weight) {
+
+        User user = userService.findUserByEmail(userEmail);

```

Comment:
Have you checked if user details already inside SecurityContext? If yes, you don't need to query it again